homeassistant/components/sonos/household_coordinator.py

```diff
@@ -0,0 +1,74 @@
+"""Class representing a Sonos household storage helper."""
+from __future__ import annotations
+
+from collections import deque
+from collections.abc import Coroutine
+import logging
+from typing import Any
+
+from pysonos import SoCo
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.debounce import Debouncer
+
+from .const import DATA_SONOS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SonosHouseholdCoordinator:
```

Comment:
Why not inherit and leverage our DataUpdateCoordinator helper? Functionality seems to be very similar.

Reply:
This class provides two different update methods where one is debounced by time, and the other via an identifier passed to the update method. It also intentionally decouples the entity updating from the API request to fetch/cache new data. And (perhaps most importantly) the update method needs to pass in an object to use in the updates. This doesn't seem possible in DUC today.

I'm open to the idea but it's looking like it would require a significant redesign of the integration to accomplish.

Reply:
To (hopefully) clarify some more, these classes are not updating from one single API endpoint. We can query any of the devices at the same API point to get the data we need. Unfortunately they don't all update at the same time, but if we carefully choose which device (endpoint) to use during each global update we can guarantee we have the latest data.

Reply:
I think we can simply inherit from DataUpdateCoordinator here and leverage the debouncer and entity callback system that the coordinator has. Everything else can stay the same.

Reply:
Another wrinkle is that we'd need to pass in a custom debouncer to override the immediate/timeout values. Both of the inherited classes are instantiated in sync context so we'd need to switch to create those debouncers before passing them into the DUC class. Am I missing an easier way to accomplish this?

Reply:
@MartinHjelmare do you think it's necessary to attempt this conversion here? I like the idea but it seems a non-trivial hurdle without refactoring of the discovery callbacks.